src/datajudge/constraints/stats.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any, Optional, Tuple
+
+import sqlalchemy as sa
+from scipy.stats import ks_2samp
+
+from .. import db_access
+from ..db_access import DataReference
+from .base import Constraint, OptionalSelections
+
+
+class KolmogorovSmirnov2Sample(Constraint):
+    def __init__(
+        self, ref: DataReference, ref2: DataReference, significance_level: float = 0.05
+    ):
+        self.significance_level = significance_level
+        super().__init__(ref, ref2=ref2)
+
+    @staticmethod
+    def calculate_2sample_ks_test(data, data2):
+        """
+        For two given lists of values calculates the Kolmogorov-Smirnov test.
+        Read more here: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kstest.html
+        """
+
+        # calculate statistic
+        statistic, p_value = ks_2samp(data, data2)
+
+        return p_value
+
+    def retrieve(
+        self, engine: sa.engine.Engine, ref: DataReference
+    ) -> Tuple[Any, OptionalSelections]:
+        return db_access.get_column(engine, ref)
+
+    def compare(
+        self, value_factual: Any, value_target: Any
+    ) -> Tuple[bool, Optional[str]]:
+        # factual and target values are the corresponding columns from our data source
```

Comment:
Again, rather superfluous as comment. Seems more to clarify yourself than to help a reader of the future.